docs/rsdl/rapid-pro-rsdl-abnf.md

```diff
@@ -6,84 +6,91 @@ title: RAPID SDL ABNF
 # RAPID Pro syntax
 
 > DRAFT
-Initial Draft. July 2020
-
+> October 2020
 
 ## Overview
 
 This grammar uses ABNF as defined by [RFC5234](https://tools.ietf.org/html/rfc5234).
 
-Note: whitespace is not (yet) reflected in the grammar.
+Note: to increase readability of the grammer
+
+- whitespace is not reflected
+- double-quoted literals are case-_sensitive_
 
 ## Syntax rules
 
 - [Model](#model)
-- [Type definition](#type-definition)
-- [Enum definition](#enum-definition)
-- [Service definition](#service-definition)
-- [Core syntax elements](#core-syntax-elements)
+- [Structured Type](#structured-type)
+- [Enumeration Type](#enumeration-type)
+- [Service](#service)
+- [Core Syntax Elements](#core-syntax-elements)
 
 ### Model
 
 ```ABNF
-model               = 1*modelElement
+model               = [ namespace ] *modelElement
 
-modelElement        = typeDefinition
-                    / serviceDefinition
-                    / enumDefinition
+namespace           = "namespace" qualifiedName
+
+modelElement        = structuredType / enumType / service
 ```
 
-### Type definition
+### Structured Type
 
 ```ABNF
-typeDefinition      = "type" "{" typeMember "}"
+structuredType      = "type" identifier "{" *typeMember "}"
 
-typeMember          = property / function ; property or bound function
+typeMember          = property / operation ; property or bound action or bound function
 
 property            = *propertyAnnotation identifier ":" typeReference
 
 propertyAnnotation  = "@key"
 
-function            = functionAnnotation identifier
-                      "(" [ parameter *("," parameter) ] ")"
-                      [ ":" typeReference ]
+typeReference       = typeName [ "?" ] / "[" typeName [ "?" ] "]"
 
-functionAnnotation  = ( "@action" / "@function" )
+typeName            = builtInType / "Edm" "." identifier / qualifiedName
 
-parameter           = identifier ":" typeReference
+builtInType         = "boolean" / "date" / "datetime" / "decimal" / "double" / "integer" / "string"
 
-typeReference       = typeName [ "?" ] / "[" typeName [ "?" ] "]"
+operation           = [ actionAnnotation ] identifier
+                      "(" [ parameter *("," parameter) ] ")"
+                      [ ":" typeReference ]
 
-typeName            = builtInType / "Edm" "." identifier / identifier
+actionAnnotation    = "@function"
```

Comment:
Should this be named `operationAnnotation` to align with `operation `

Secondly what is the reason for removing the `@action`  When we discussed this a while ago we said everything is a function by default and actions are "opt-in"  I also like to keep the `@function` just for symetry and say in the absence of an annotation it defaults to `@function`.